cms/src/main/java/uk/nhs/digital/ps/modules/FullTaxonomyModule.java

```diff
@@ -0,0 +1,153 @@
+package uk.nhs.digital.ps.modules;
+
+import org.apache.jackrabbit.value.StringValue;
+import org.onehippo.repository.events.HippoWorkflowEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.jcr.Value;
+
+public class FullTaxonomyModule extends AbstractDaemonModule {
+
+    private static final Logger log = LoggerFactory.getLogger(FullTaxonomyModule.class);
+
+    public static final String HIPPO_CLASSIFIABLE_PATH = "/hippo:namespaces/publicationsystem/publication/editor:templates/_default_/classifiable";
+    public static final String TAXONOMY_NAME_PROPERTY = "essentials-taxonomy-name";
+    public static final String FULL_TAXONOMY_PROPERTY = "common:FullTaxonomy";
+
+    private Node taxonomyTreeNode;
+
+    @Override
+    public void initialize(final Session session) {
+        super.initialize(session);
+
+        this.taxonomyTreeNode = getTaxonomyTreeNode();
+    }
+
+    @Override
+    protected void handleCommitEvent(HippoWorkflowEvent event) {
+
+        try {
+            Node node = getSession().getNode(event.subjectPath());
+
+            // don't do the published ones, these will get populated when the document is published
+            // also don't need to bother with node with no taxonomy
+            streamChildNodes(node)
+                .filter((document) -> !STATE_PUBLISHED.equals(getRepoValue(() -> document.getProperty("hippostd:state").getString())))
+                .filter((document) -> getRepoValue(() -> document.hasProperty("hippotaxonomy:keys")))
+                .forEach(this::createFullTaxonomyProperty);
+
+            getSession().save();
+        } catch (RepositoryException re) {
+            throw new RuntimeException(re);
+        }
+    }
+
+    private void createFullTaxonomyProperty(Node document) {
+        Set<String> fullTaxonomyKeys = getFullTaxonomyKeys(document);
+        Value[] fullTaxonomy = fullTaxonomyKeys.stream()
+            .map(StringValue::new)
+            .toArray(Value[]::new);
+
+        try {
+            document.setProperty(FULL_TAXONOMY_PROPERTY, fullTaxonomy);
+        } catch (RepositoryException re) {
+            log.error("Failed to set full taxonomy on node: " + document);
```

Comment:
Should the exception details be passed as well as the document?  Does it need to be thrown?

Reply:
Good spot!
Don't want to throw as we are quite low down and don't want to interupt the triggering event but should definitley be logging the exception, have updated. 🙂